kubernetes/config-maps/nginx/ssl.conf

```diff
@@ -0,0 +1,26 @@
+# Disable SSLv3
+ssl_protocols              TLSv1 TLSv1.1 TLSv1.2;
+
+# Enable server-side protection
+ssl_prefer_server_ciphers  on;
+ssl_ciphers                "EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH";
+
+# Specify curve P-384
+ssl_ecdh_curve             secp384r1;
+
+# Improve ssl performence
+ssl_session_cache          shared:SSL:10m;
+ssl_session_timeout        5m;
+
+# Enable OCSP stapling
+ssl_stapling               on;
+ssl_stapling_verify        on;
+
+# Enable HSTS
+add_header Strict-Transport-Security "max-age=63072000; includeSubdomains";
+
+# Advanced settings
+server_tokens off;
+add_header X-XSS-Protection "1; mode=block";
+add_header X-Frame-Options DENY;
+add_header X-Content-Type-Options nosniff;
```

Comment:
These options also have nothing to do with SSL. Please move them somewhere else.

Also, please add some explanations. At least links to their definitions:

```
# Enable XSS protection for ancient browsers that don't support CSP.
# See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-XSS-Protection
add_header X-XSS-Protection "1; mode=block";

# Prevent rendering our pages inside HTML frames.
# We don't have frames (and we don't really serve many web pages) so there's no harm in enabling it.
# See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options
add_header X-Frame-Options DENY;

# Enforce strict interpretation of `Content-Type` for files included in `<script>` and `<style>` HTML tags.
# Again, relevant only for webpages but there's no harm in having it enabled.
# See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Content-Type-Options
add_header X-Content-Type-Options nosniff;
```

Reply:
Also, keep this in mind: [Nginx add_header configuration pitfall ](https://blog.g3rt.nl/nginx-add_header-pitfall.html).

Reply:
Done.